hphp/runtime/vm/jit/vasm-simplify.cpp

```diff
@@ -375,22 +375,35 @@ bool simplify(Env& env, const copyargs& inst, Vlabel b, size_t i) {
 }
 
 bool simplify(Env& env, const movzlq& inst, Vlabel b, size_t i) {
-  if (arch() != Arch::X64) return false;
-  auto const def_op = env.def_insts[inst.s];
-
-  // Check if `inst.s' was defined by an instruction with Vreg32 operands, or
-  // movzbl{} in particular (which lowers to a movl{}).
-  if (width(def_op) != Width::Long &&
-      def_op != Vinstr::movzbl) {
-    return false;
+  switch (arch()) {
```

Comment:
Can you just do `if (arch() == Arch::PPC64) return false;` instead?

Might be time for me to split up vasm-simplify per-architecture.